### PR TITLE
Fix switch

### PR DESCRIPTION
**Pull request by troibe**
**Why:**
Up to this point the switch on the demo page was not working #6 

**What:**
Added the switch status to the model

**Testing:**
|Before|After| 
|---|---|
|Fix-Switch-before|Fix-Switch-after|

The slow response time should only be due to lots of other programs running in the background.

### CODE REVIEW

File: sample-app-jsaddle/Material/Snackbar.hs [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+
+
+{-| Snackbar view function
+-}
+snackbar :: Config msg -> Queue msg -> Miso.View msg
+snackbar (config_@Config { additionalAttributes=additionalAttributes }) (queue@Queue { messages=messages, nextMessageId=nextMessageId }) =
+    let
+        ( currentMessageId, currentMessage ) =
+            Maybe.listToMaybe messages
+                |> Maybe.maybe Nothing (\x -> Just (Bifunctor.second Just x))
```
Comment (adrianomelo) [inline sample-app-jsaddle/Material/Snackbar.hs]:
```suggestion
                |> fmap (Bifunctor.second Just)
```

File: sample-app-jsaddle/Material/Snackbar.hs [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+                Miso.button_
+                    (Maybe.mapMaybe id
+                        [ actionButtonCs
+                        , actionButtonClickHandler messageId message_
+                        ]
+                    )
+                    [ Miso.text (Miso.String.toMisoString actionButtonLabel) ]
+            )
+        )
+        actionButton
```
Comment (adrianomelo) [inline sample-app-jsaddle/Material/Snackbar.hs]:
```suggestion
    fmap
        (\actionButtonLabel ->
            (Miso.button_
                (Maybe.mapMaybe id
                    [ actionButtonCs
                    , actionButtonClickHandler messageId message_
                    ]
                )
                [ Miso.text (Miso.String.toMisoString actionButtonLabel) ]
            )
        )
        actionButton
```

File: sample-app-jsaddle/Material/Snackbar.hs [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+                    nodes
+                )
+
+        Nothing ->
+            Nothing
+
+
+actionIconClickHandler :: MessageId -> Message msg -> Maybe (Miso.Attribute msg)
+actionIconClickHandler messageId (Message { onActionIconClick=onActionIconClick }) =
+    Maybe.maybe Nothing (\x->Just (Miso.onClick $ (|>) messageId x)) onActionIconClick
```
Comment (adrianomelo) [inline sample-app-jsaddle/Material/Snackbar.hs]: Another possibility:

```suggestion
    fmap (\x -> (Miso.onClick $ (|>) messageId x)) onActionIconClick
```

**Review by adrianomelo — APPROVED** [review]

**Comment by adrianomelo** [comment]
I actually added the review of the Snackbar into this PR.

Thanks for inviting me to be an admin of the project. :+1: 

I believe some of the open pull requests are independent. I suggest creating PRs with the isolated commits to make it easier to review.

**Comment by troibe** [comment]
Thanks for the reviews!
I'm always happy to learn and improve my Haskell skills.
Since I was working with a rebase workflow in my last internship instead of the more traditional merge workflow I stacked the pull requests.
For normal git usage you would create independent pull requests and solve merge conflicts after the fact right?
In this case the only conflicts occurring would be on Main.hs...

**Comment by adrianomelo** [comment]
>  For normal git usage you would create independent pull requests and solve merge conflicts after the fact right?

Yes, the way I usually do is keeping different branches with independent changes. You probably won't have conflicts or they should be easy to fix in this case. As you said, the only common file is `Main.hs`, and, for most of the PRs, you are only adding lines.

It is tricky when you have so many pull requests, but in this case, It would be better to combine multiple components instead of having the same commits on multiple pull requests. (from my point of view)

**Pull request merged**